hardhat-tests/test/internal/hardhat-network/stack-traces/test.ts

```diff
@@ -529,7 +533,8 @@ async function runTest(
 
     compareConsoleLogs(logger.lines, tx.consoleLogs);
 
-    const vmTraceDecoder = (provider as any)._vmTraceDecoder as VmTraceDecoder;
+    // @ts-ignore - this is a private field but don't `as any` for go-to-def to work
+    const vmTraceDecoder = provider._vmTraceDecoder as VmTraceDecoder;
```

Comment:
I'd rather not use `@ts-ignore` anywhere (we never do in the Hardhat codebase), since it's a slippery slope. For this particular pattern, do this instead:

```js
    // eslint-disable-next-line @typescript-eslint/dot-notation
    const vmTraceDecoder = provider["_vmTraceDecoder"] as VmTraceDecoder;
```

Basically: typescript lets you access private properties using `["foo"]` but not `.foo` (no idea why), but we have an eslint rule to prefer dot-notation, so we need to disable it for this line.

The `as VmTraceDecoder` is still needed, but it _shouldn't_. I don't know why; it seems like a typescript bug. I tried to reproduce it in the playground but couldn't, so let's just keep the casting.

Reply:
Sure, makes sense! I wonder if that may have been fixed in the newer TypeScript versions? I think we're still using Hardhat's, which is ^5.0.4.

Reply:
No, I tried updating typescript and got the same result :confused: